app/main.py

```diff
@@ -1 +1,58 @@
-#  write your code here
+class Animal:
+    animals = []
+
+    def __init__(
+            self,
+            name: str,
+            appetite: int,
+            is_hungry: bool = True
+    ) -> None:
+
+        self.name = name
+        self.is_hungry = is_hungry
+        self.appetite = appetite
+        self.animals.append(self)
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        food_points = 0
+
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            food_points += self.appetite
+
+            return food_points
+
+        return 0
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=3)
+        self.is_hungry = is_hungry
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=7)
```

Comment:
You should pass `is_hungry` in the `__init__` as well, add this for `Dog` and `Cat`.

Reply:
![image](https://github.com/mate-academy/py-feed-animals/assets/140983258/df038dc8-c603-4269-a872-88d1e950d0dd)

but it contradicts the condition, though

Reply:
figured it out